src/__tests__/addStateHandlers.coffee

```diff
@@ -226,68 +133,51 @@ describe 'addStateHandlers', ->
     rerender <Component />
     expect(getInitial).toHaveBeenCalledTimes 1
 
-  test 'state values are dependencies when using dependencies callback', ->
-    Inner = flow(
-      addStateHandlers
-        x: 0
-      ,
-        incrementXbyY: ({x}, {y}) -> ->
-          x: x + y
-      ,
-        (oldProps, newProps) -> oldProps.y isnt newProps.y
-    ,
-      ({x, incrementXbyY}) ->
-        <div>
-          <div data-testid="d">{x}</div>
-          <button onClick={incrementXbyY}>incrementXbyY</button>
-        </div>
-    )
-    Outer = flow(
-      addState 'y', 'setY', 1
-      ({y, setY}) ->
+  test 'handles multiple updates on same render', ->
+    Component = flow(
+      addStateHandlers {x: 1}, incrementX: ({x}) -> -> x: x + 1
+      ({x, incrementX, testId}) ->
         <div>
-          <Inner y={y} />
-          <button onClick={-> setY 2}>update Y</button>
+          <div data-testid={testId}>{x}</div>
+          <button
+            onClick={->
+              incrementX()
+              incrementX()
+            }
+          >
+            trigger multiple updates
+          </button>
         </div>
     )
-    {getByTestId, getByText} = render <Outer />
-    expect(getByTestId 'd').toHaveTextContent '0'
-    fireEvent.click getByText /incrementXbyY/
-    expect(getByTestId 'd').toHaveTextContent '1'
-    fireEvent.click getByText /incrementXbyY/
-    expect(getByTestId 'd').toHaveTextContent '2'
-    fireEvent.click getByText /update Y/
-    fireEvent.click getByText /incrementXbyY/
-    expect(getByTestId 'd').toHaveTextContent '4'
-  test 'state values are dependencies when using dependencies array', ->
-    Inner = flow(
+    testId = 'multiple-updates-on-same-render'
+    {getByTestId, getByText} = render <Component testId={testId} />
+    expect(getByTestId testId).toHaveTextContent '1'
+
+    fireEvent.click getByText /trigger multiple updates/
+    expect(getByTestId testId).toHaveTextContent '3'
+  test "doesn't care if new state keys get added by handlers", ->
+    Component = flow(
       addStateHandlers
-        x: 0
+        x: 1
       ,
-        incrementXbyZ: ({x}, {z}) -> ->
-          x: x + z
-      ,
-        ['z']
-      ({x, incrementXbyZ}) ->
-        <div>
-          <div data-testid="e">{x}</div>
-          <button onClick={incrementXbyZ}>incrementXbyZ</button>
-        </div>
-    )
-    Outer = flow(
-      addState 'z', 'setZ', 1
-      ({z, setZ}) ->
+        incrementX: ({x}) -> ->
+          x: x + 1
+          hasIncremented: true
+      ({x, incrementX, testId, hasIncremented}) ->
```

Comment:
So the new state props become available to the rest of the chain? That seems hard/impossible to support in TS (which I'm fine with, since why would you ever really do this).

Reply:
Yup that makes sense to me, it's not a very "strongly typed" thing to do

I guess it might work (if you really felt compelled to not just initialize the state key to `undefined`) to cast the initial state argument with an optional property eg:
```
addStateHandlers(
  {x: 1} as {x: number, hasIncremented?: boolean},
  ...
)
```
I'm assuming then Typescript would expose a `hasIncremented: boolean | undefined` prop type to the rest of the chain?